lib/_http_client.js

```diff
@@ -30,10 +30,17 @@ function ClientRequest(options, cb) {
 
   var agent = options.agent;
   var defaultAgent = options._defaultAgent || Agent.globalAgent;
+
+  if (agent != null && agent !== false && !(agent instanceof Agent.Agent)) {
+    throw new Error(
+       'Agent option must be an instance of http.Agent, undefined or false. ' +
+       `Received type: ${typeof agent} instead`
+    );
+  }
+
   if (agent === false) {
     agent = new defaultAgent.constructor();
-  } else if ((agent === null || agent === undefined) &&
-             typeof options.createConnection !== 'function') {
+  } else if (agent == null && typeof options.createConnection !== 'function') {
     agent = defaultAgent;
   }
```

Comment:
It's possible for agent to be null at this point, so I could do an else if here and check that it's not null and that it's not an instance of Agent.Agent. 

I went this route as it verifies that we are dealing with appropriate types before this if statement and because the else if logic seemed slightly verbose.

I am okay with putting the extra checks on the else if statement though if it's preferred.